core/renderers/measurables/rows.js

```diff
@@ -0,0 +1,441 @@
+ */
+Blockly.blockRendering.TopRow.prototype.measure = function() {
+  this.width = 0;
+  this.height = 0;
+  for (var e = 0; e < this.elements.length; e++) {
+    var elem = this.elements[e];
+    this.width += elem.width;
+    if (!(elem.isSpacer())) {
+      if (elem.type == 'hat') {
+        this.startY = elem.startY;
+        this.height = this.height + elem.height;
+      }
+      this.height = Math.max(this.height, elem.height);
+    }
+  }
+  this.widthWithConnectedBlocks = this.width;
+};
+
+/**
+ * An object containing information about what elements are in the bottom row of
+ * a block as well as spacing information for the top row.
+ * Elements in a bottom row can consist of corners, spacers and next connections.
+ * @param {!Blockly.BlockSvg} block The block for which this represents the
+ *     bottom row.
+ * @package
+ * @constructor
+ * @extends {Blockly.blockRendering.Row}
+ */
+Blockly.blockRendering.BottomRow = function(block) {
+  Blockly.blockRendering.BottomRow.superClass_.constructor.call(this);
+  this.type = 'bottom row';
+
+  /**
+   * Whether the block has a next connection.
+   * @package
+   * @type {boolean}
+   */
+  this.hasNextConnection = !!block.nextConnection;
+
+  /**
+   * The next connection on the block, if any.
+   * TODO: Should this be the connection measurable instead?  It would add some
+   * indirection but would mean we aren't mixing connections and connection
+   * measurables.
+   * @package
+   * @type {Blockly.RenderedConnection}
+   */
+  this.connection = block.nextConnection;
+
+  /**
+   * The amount that the bottom of the block extends below the horizontal edge,
+   * e.g. because of a next connection.  Must be non-negative (see #2820).
+   * @package
+   * @type {number}
+   */
+  this.overhangY = 0;
+
+  var followsStatement =
+      block.inputList.length &&
+      block.inputList[block.inputList.length - 1].type == Blockly.NEXT_STATEMENT;
+  this.hasFixedWidth = followsStatement && block.getInputsInline();
+
+  // This is the minimum height for the row. If one of its elements has a greater
+  // height it will be overwritten in the compute pass.
+  if (followsStatement) {
+    this.height = Blockly.blockRendering.constants.LARGE_PADDING;
+  } else {
+    this.height = this.notchShape.height;
+  }
+};
+goog.inherits(Blockly.blockRendering.BottomRow,
+    Blockly.blockRendering.Row);
+
+/**
+ * Convenience method to get the measurable representing the next
+ * connection, if one exists.
+ * @return {Blockly.blockRendering.NextConnection} The measurable that
+ *     represents the next connection on this block, or null.
+ * @package
+ */
+Blockly.blockRendering.BottomRow.prototype.getNextConnection = function() {
+  if (this.hasNextConnection) {
+    return /** @type {Blockly.blockRendering.NextConnection} */ (this.elements[2]);
+  }
+  return null;
+};
+
+/**
+ * @override
+ */
+Blockly.blockRendering.BottomRow.prototype.measure = function() {
+  this.width = 0;
+  this.height = 0;
```

Comment:
There is some logic for Bottom row for the minimum height of the row. Doesn't the reset of width and height need to take that into account?

Reply:
Good catch.  Fixed.